### PR TITLE
feat: MLIBZ-2474 Add @NonNull and @Nullable annotations for DataStore and UserStore

### PR DESCRIPTION
**Pull request by yuliya-guseva**
#### Description
Add `@NonNull` and `@Nullable` annotations for DataStore and UserStore

#### Changes
Added `@NonNull` and `@Nullable` annotations to `DataStore`, `BaseDataStore`, `RealmCache`, `ICache`, `UserStore/BaseUserStore` classes.

#### Tests
Instrumented

### CODE REVIEW

File: android-lib/src/main/java/com/kinvey/android/store/DataStore.java [hunk 1]
```diff
@@ -709,7 +713,7 @@ public void pull(KinveyPullCallback callback) {
      * @param isAutoPagination true if auto-pagination is used
      * @param callback KinveyPullCallback
      */
-    public void pull(Query query, boolean isAutoPagination, KinveyPullCallback callback) {
+    public void pull(@Nullable Query query, boolean isAutoPagination, @NonNull KinveyPullCallback callback) {
         Preconditions.checkNotNull(client, "client must not be null");
```
Comment (vinaygahlawat) [inline android-lib/src/main/java/com/kinvey/android/store/DataStore.java]: Why is the `Query` allowed to be `null` here?

Comment (yuliya-guseva) [reply android-lib/src/main/java/com/kinvey/android/store/DataStore.java]: It's old logic. Inside the method, the SDK checks the query is it's null then the new empty query will be created.

File: java-api-core/src/com/kinvey/java/cache/ICache.java [hunk 2]
```diff
@@ -98,19 +109,21 @@
...
-    T getFirst(Query q);
+    @Nullable
+    T getFirst(@Nonnull Query q);
 
     /**
      * gets the count of matching elements from the cache
      * @return first the count of matching elements from the cache
      */
-    long count(Query q);
+    long count(@Nullable Query q);
```
[the remark this answers is not in the document]
Comment (yuliya-guseva) [reply java-api-core/src/com/kinvey/java/cache/ICache.java]: In the method implementation, we check the query to null, so I think that that query can be null. If the query is null, the method will return a count of all items. 
https://github.com/Kinvey/java-library/blob/6a26693fe02e244058ed383f64dc404ffba78da0/android-lib/src/main/java/com/kinvey/android/cache/RealmCache.java#L458

[the remark this answers is not in the document]
Comment (vinaygahlawat) [reply java-api-core/src/com/kinvey/java/cache/ICache.java]: Ok, so there is no override method of `count()` with no `Query` parameter?

[the remark this answers is not in the document]
Comment (yuliya-guseva) [reply java-api-core/src/com/kinvey/java/cache/ICache.java]: `DataStore` has only method `count` without any parameters, but the SDK uses `count` from RealmCache with `query` for getting count sync requests in the sync collection.

**Pull request merged**